tools/board-runner/src/opentitan.rs

```diff
@@ -0,0 +1,423 @@
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/mpu_stack_growth/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let mut p = opentitan_flash(&app).unwrap();
+
+    p.exp_string("This test should recursively add stack frames until exceeding")?;
+    p.exp_string("panicked at 'Process mpu_stack_growth had a fault'")?;
+    p.exp_string("Store/AMO access fault")?;
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn opentitan_mpu_walk_region() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/mpu_walk_region/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let mut p = opentitan_flash(&app).unwrap();
+
+    p.exp_string("MPU Walk Regions")?;
+    p.exp_string("Walking flash")?;
+    p.exp_string("Will overrun")?;
+    p.exp_string("panicked at 'Process mpu_walk_region had a fault'")?;
+
+    Ok(())
+}
+
+fn opentitan_multi_alarm_test() -> Result<(), Error> {
+    let app = format!(
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/multi_alarm_test/build/rv32imc/rv32imc.0x20030080.0x10005000.tbf"
+    );
+    let _p = opentitan_flash(&app).unwrap();
+
+    println!("Make sure the LEDs are blinking");
+
+    let timeout = time::Duration::from_secs(10);
+    thread::sleep(timeout);
+
+    Ok(())
+}
+
+pub fn all_opentitan_tests() {
+    println!("Tock board-runner starting...");
+    println!();
+    println!("Running opentitan tests...");
+    opentitan_c_hello().unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+    opentitan_blink().unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+    opentitan_c_hello_and_printf_long()
+        .unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+    opentitan_recv_short_and_recv_long()
+        .unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+    opentitan_blink_and_c_hello_and_buttons()
+        .unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+    opentitan_console_recv_short().unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+    opentitan_console_timeout().unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+
+    // Disabled by default.
+    // Requires:
+    //    STACK_SIZE       = 2048
+    //    APP_HEAP_SIZE    = 4096
+    //    KERNEL_HEAP_SIZE = 2048
+    // opentitan_malloc_test1().unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+
+    // Disabled by default.
+    // Requires:
+    //    STACK_SIZE       = 2048
+    //    APP_HEAP_SIZE    = 4096
+    //    KERNEL_HEAP_SIZE = 2048
+    // opentitan_stack_size_test1().unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
+
+    // Disabled by default.
+    // Requires:
+    //    STACK_SIZE       = 2048
+    //    APP_HEAP_SIZE    = 4096
+    //    KERNEL_HEAP_SIZE = 2048
+    // opentitan_stack_size_test2().unwrap_or_else(|e| panic!("opentitan job failed with {}", e));
```

Comment:
Why do these require different app/kernel heap sizes on OpenTitan, but not on other platforms?

Reply:
I'm not sure. Maybe it's related to app addresses not exactly matching what the kernel expects?

I did try using a few different addresses, but that didn't seem to help.